File: Elearn-ApplicationTier/Elearn.Shared/Models/Question.cs
namespace Elearn.Shared.Models;

public class Question
{
    public long Id { get; set; }
    public string Title { get; set; }
    public string Body { get; set; }
    
    public string Description { get; set; }
    public string Url { get; set; }
    public DateTime CreationDate { get; set; }

    public long CorrectAnswerId { get; set; }
    
    public Student Author { get; set; }

    public Question(long id, string title, string body, string description, string url, DateTime creationDate, long correctAnswerId, Student author)
    {
        Id = id;
        Title = title;
        Body = body;
        Description = description;
        Url = url;
        CreationDate = creationDate;
        CorrectAnswerId = correctAnswerId;
        Author = author;
    }

    public Question(string title, string body, string description, string url, DateTime creationDate, long correctAnswerId, Student author)
    {
        Title = title;
        Body = body;
        Description = description;
        Url = url;
        CreationDate = creationDate;
        CorrectAnswerId = correctAnswerId;
        Author = author;
    }

    public Question()
    {
    }
    
    public override string ToString()
    {
        return $"Id: {Id}, Title: {Title}, Body: {Body}, Url: {Url}, CreationDate: {CreationDate}, CorrectAnswerId: {CorrectAnswerId}, Author: {Author}";
    }
}

File: Elearn-DataTier/src/main/java/via/dk/elearn/models/Question.java
package via.dk.elearn.models;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import javax.persistence.*;
import java.time.LocalDate;

@Entity
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Question {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(name = "title", length = 50)
    private String title;

    @Column(name = "description", length = 80)
    private String description;

    @Lob
    @Column(name = "body")
    private String body;

    @Column(name = "url")
    private String url;

    @Column
    private LocalDate date;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "course_id")
    @OnDelete(action = OnDeleteAction.CASCADE)
    private Course course;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "user_id")
    @OnDelete(action = OnDeleteAction.CASCADE)
    private Student student;

    public String toString() {
        return "Question: " + title + " " + body + " " + url;
    }

    public Question(String title, String description, String body, String url, LocalDate date, Course course, Student student) {
        this.title = title;
        this.description = description;
        this.body = body;
        this.url = url;
        this.date = date;
        this.course = course;
        this.student = student;
    }
}
